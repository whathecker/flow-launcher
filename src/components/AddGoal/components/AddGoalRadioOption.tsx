import React from "react";
import { Text, Touchable } from "../../Themed";
import { Container, Typography } from "../../../styles";
import { Reminder } from "../../../types/entity";

type AddGoalRadioOptionProps = {
  pressHandler: () => void;
  name: Reminder;
  displayText: string;
  activeValue: string;
};

const AddGoalRadioOption: React.FC<AddGoalRadioOptionProps> = ({
  displayText,
  name,
  activeValue,
  pressHandler,
}: AddGoalRadioOptionProps) => {
  const isBtnActive = name === activeValue ? true : false;
  return (
    <Touchable
      onPress={pressHandler}
      lightColor="#ffffff"
      darkColor="#ffffff"
      style={{
        ...Container.centerAligned,
        width: "100%",
        height: 50,
        paddingTop: 10,
        paddingBottom: 10,
        marginTop: 10,
        marginBottom: 10,
        borderRadius: 10,
        borderWidth: isBtnActive ? 2 : 1,
        borderColor: isBtnActive ? "#0FA858" : "#636262",
        borderStyle: "solid",
        overflow: "hidden",
      }}
    >
      <Text
        lightColor={isBtnActive ? "#0FA858" : "#636262"}
        darkColor={isBtnActive ? "#0FA858" : "#636262"}
        style={{ ...Typography.p }}
      >
        {displayText}
      </Text>
    </Touchable>
  );
};

export default AddGoalRadioOption;
