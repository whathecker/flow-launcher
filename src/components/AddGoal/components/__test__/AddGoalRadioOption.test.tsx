import * as React from "react";
import renderer from "react-test-renderer";
import AddGoalRadioOption from "../AddGoalRadioOption";

describe("Test AddGoalRadioOption component", () => {
  it("Should renders inactive AddGoalRadioOption correctly", () => {
    const tree = renderer
      .create(
        <AddGoalRadioOption
          name="daily"
          displayText="Every Day"
          activeValue="three_days"
          pressHandler={() => {
            return;
          }}
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("Should renders active AddGoalRadioOption correctly", () => {
    const tree = renderer
      .create(
        <AddGoalRadioOption
          name="seven_days"
          displayText="Every Day"
          activeValue="seven_days"
          pressHandler={() => {
            return;
          }}
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
