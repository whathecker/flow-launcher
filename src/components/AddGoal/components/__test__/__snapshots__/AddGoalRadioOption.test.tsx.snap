// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test AddGoalRadioOption component Should renders active AddGoalRadioOption correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "borderColor": "#0FA858",
      "borderRadius": 10,
      "borderStyle": "solid",
      "borderWidth": 2,
      "flexDirection": "row",
      "height": 50,
      "justifyContent": "center",
      "marginBottom": 10,
      "marginTop": 10,
      "opacity": 1,
      "overflow": "hidden",
      "paddingBottom": 10,
      "paddingTop": 10,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#0FA858",
        },
        Object {
          "fontFamily": "Forum-Regular",
          "fontSize": 20,
        },
      ]
    }
  >
    Every Day
  </Text>
</View>
`;

exports[`Test AddGoalRadioOption component Should renders inactive AddGoalRadioOption correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "borderColor": "#636262",
      "borderRadius": 10,
      "borderStyle": "solid",
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 50,
      "justifyContent": "center",
      "marginBottom": 10,
      "marginTop": 10,
      "opacity": 1,
      "overflow": "hidden",
      "paddingBottom": 10,
      "paddingTop": 10,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#636262",
        },
        Object {
          "fontFamily": "Forum-Regular",
          "fontSize": 20,
        },
      ]
    }
  >
    Every Day
  </Text>
</View>
`;
